src/log/access_log.cc

```diff
@@ -244,21 +238,23 @@ HierarchyLogEntry::notePeerWrite()
 void
 HierarchyLogEntry::startPeerClock()
 {
-    if (!firstConnStart_.tv_sec)
-        firstConnStart_ = current_time;
+    if (peerTimer.running()) {
+        debugs(46, 5, "already running");
+        return;
+    }
+    peerTimer.resume();
 }
 
 void
-HierarchyLogEntry::stopPeerClock(const bool force)
+HierarchyLogEntry::stopPeerClock()
 {
-    debugs(46, 5, "First connection started: " << firstConnStart_ <<
-           ", current total response time value: " << (totalResponseTime_.tv_sec * 1000 +  totalResponseTime_.tv_usec/1000) <<
-           (force ? ", force fixing" : ""));
-    if (!force && totalResponseTime_.tv_sec != -1)
+    if (!peerTimer.running()) {
+        debugs(46, 5, "already stopped");
         return;
```

Comment:
> When I added these checks, I was a little worried that peerTimer may ... fail  ... to pause() because it was previously resume() twice.

I do not think any code can distinguish the following two cases:

1. resume() was correctly called twice (e.g., the second call was correctly done by a sub-task or even an asynchronous partially overlapping task that uses the same timer/measurement).
2. resume() was incorrectly called twice (i.e. a bug)

If we want to support use case 1 (and I think we do!), then we cannot add these kind of checks AFAICT.